compiler/rustc_mir/src/const_eval/mod.rs

```diff
@@ -139,14 +203,15 @@ fn const_to_valtree_inner<'tcx>(
 pub(crate) fn destructure_const<'tcx>(
     tcx: TyCtxt<'tcx>,
     param_env: ty::ParamEnv<'tcx>,
-    val: &'tcx ty::Const<'tcx>,
+    val: ConstValue<'tcx>,
+    ty: Ty<'tcx>,
 ) -> mir::DestructuredConst<'tcx> {
```

Comment:
This code should disappear eventually, right? When the pattern matching code was switched to ValTrees? A `ValTree` is essentially pre-destructed; this code here turns it into a regular `Operand` again only to then destruct that, which makes little sense.

OTOH I see the pattern matching code is already being changed to some extend here. What is even still using `destructure_const`?

Reply:
> OTOH I see the pattern matching code is already being changed to some extend here. What is even still using `destructure_const`?

To keep the diff as small as possible (not necessarily in code, but in behaviour). The next step would be to make pattern matching treat all non-valtree consts as opaque

Reply:
I see, makes sense -- I hope we won't lose track of this. :)

Reply:
Is the same true for `op_to_const`? That function is just such a disaster...